src/qengine/state/state.cpp

```diff
@@ -87,28 +87,47 @@ void QEngineCPU::SetQuantumState(Complex16* inputState)
  *
  * A fundamental operation used by almost all gates.
  */
-void QEngineCPU::Apply2x2(bitCapInt offset1, bitCapInt offset2, const Complex16* mtrx, const bitLenInt bitCount,
+void QEngineCPU::Apply2x2(bitCapInt offset1, bitCapInt offset2, const complex* mtrx, const bitLenInt bitCount,
     const bitCapInt* qPowersSorted, bool doCalcNorm)
 {
-    Complex16 nrm = Complex16((bitCount == 1) ? (1.0 / runningNorm) : 1.0, 0.0);
     int numCores = GetConcurrencyLevel();
+    double nrm = 1.0 / runningNorm; 
+#if ENABLE_AVX  
+    ComplexUnion mtrxCol1;
```

Comment:
Given that this type is only used here in this function, either:
  A. Move the type into this function (or the split ENABLE_AVX variant of Apply2x2) as a simple union type, declared at the top of the function.
  B. Move the type into this file, declared at the top

Not necessary to have it in the complex16x2simd.hpp header, since it’s of such limited and localized utility.

Reply:
Done, including type alias definition.